src/transformers/modeling_prophetnet.py

```diff
@@ -0,0 +1,1461 @@
+
+        encoder_states, all_attentions = [], []
+        for encoder_layer in self.layers:
+            if self.output_hidden_states:
+                encoder_states.append(x)
+            x, attn = encoder_layer(x, attention_mask, output_attentions=output_attentions)
+            if output_attentions:
+                all_attentions.append(attn)
+
+        if self.output_hidden_states:
+            encoder_states.append(x)
+
+        # T x B x C -> B x T x C
+        encoder_states = [hidden_state.transpose(0, 1) for hidden_state in encoder_states]
+        x = x.transpose(0, 1)
+        results = (x, encoder_states, all_attentions)
+        return results
+
+    def ngram_attention_bias(length, num_skip):
+        bias_result = []
+        for n_skip in range(num_skip):
+            bias_n_skip = []
+            for i in range(length):
+                bias_this = [float('-inf')] * (2 * length)
+                bias_this[length + i] = 0
+                first_k = i - n_skip
+                first_k = first_k if first_k > 0 else 0
+                for j in range(first_k + 1):
+                    bias_this[j] = 0
+                bias_n_skip.append(bias_this)
+            bias_result.append(bias_n_skip)
+        return torch.from_numpy(np.array(bias_result, dtype=np.float32))
+
+def softmax(x, dim, onnx_trace=False):
+    if onnx_trace:
+        return F.softmax(x.float(), dim=dim)
+    else:
+        return F.softmax(x, dim=dim, dtype=torch.float32)
+
+class NgramMultiheadAttention(nn.Module):
+    def __init__(self, embed_dim, num_heads, kdim=None, vdim=None, dropout=0., bias=True,
+                 add_bias_kv=False, add_zero_attn=False, self_attention=False,
+                 encoder_decoder_attention=False, ngram=2, num_buckets=32, relative_max_distance=128):
+        super().__init__()
+        self.embed_dim = embed_dim
+        self.kdim = kdim if kdim is not None else embed_dim
+        self.vdim = vdim if vdim is not None else embed_dim
+        self.qkv_same_dim = self.kdim == embed_dim and self.vdim == embed_dim
+
+        self.num_buckets = num_buckets
+        self.relative_max_distance = relative_max_distance
+        self.num_heads = num_heads
+        self.dropout = dropout
+        self.head_dim = embed_dim // num_heads
+        self.ngram = ngram
+
+        assert self.head_dim * num_heads == self.embed_dim, "embed_dim must be divisible by num_heads"
+        self.scaling = self.head_dim ** -0.5
+
+        self.self_attention = self_attention
+        self.encoder_decoder_attention = encoder_decoder_attention
+
+        assert not self.self_attention or self.qkv_same_dim, 'Self-attention requires query, key and ' \
+                                                             'value to be of the same size'
+
+        self.relative_linear = nn.Linear(embed_dim, num_buckets * num_heads)
+        if self.qkv_same_dim:
+            self.in_proj_weight = nn.Parameter(torch.Tensor(3 * embed_dim, embed_dim))
```

Comment:
don't think we need `torch.Tensor` for `nn.Parameter` -> `nn.Parameter(3 * embed_dim, embed_dim)` should be good no?